packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -269,6 +269,19 @@ function getNearestShadowAncestor(vm: VM): VM | null {
     return ancestor;
 }
 
+export function createVMContext() {
```

Comment:
this is just abstracting a segment of the vm creation process so we can create more in dynamic.

Note: I'm not 100% sure if the `ctx` is cached somewhere, I believe it is only used inside templates, in which case it doesn't matter since the template is algo going to be replaced.

Note 2: what happen if you swap ctors, but both resolves to the same `template`? jejejeje